examples/pytorch/text-generator/README.md

```diff
@@ -1,3 +1,304 @@
 
-Please refer to the [tutorial](https://docs.cortex.dev/iris-classifier) to see how to deploy an example with Cortex.
+This example shows how to deploy a text generator using a GPT-2 model from Hugging Face's transformers library.
+
+<br>
+
+## Implement your predictor
+
+1. Create a Python file named `predictor.py`.
+2. Define a Predictor class with a constructor that loads and initializes the model.
+3. Add a predict function that will accept a payload and return the generated text.
+
+```python
+# predictor.py
+
+import torch
+from transformers import GPT2Tokenizer, GPT2LMHeadModel
+
+
+class PythonPredictor:
+    def __init__(self, config):
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        print(f"using device: {self.device}")
+        self.tokenizer = GPT2Tokenizer.from_pretrained("gpt2")
+        self.model = GPT2LMHeadModel.from_pretrained("gpt2").to(self.device)
+
+    def predict(self, payload):
+        input_length = len(payload["text"].split())
+        tokens = self.tokenizer.encode(payload["text"], return_tensors="pt").to(self.device)
+        prediction = self.model.generate(tokens, max_length=input_length + 20, do_sample=True)
+        return self.tokenizer.decode(prediction[0])
+```
+
+Here are the complete [Predictor docs](../../../docs/deployments/syncapi/predictors.md).
+
+<br>
+
+## Specify your Python dependencies
+
+Create a `requirements.txt` file to specify the dependencies needed by `predictor.py`. Cortex will automatically install them into your runtime once you deploy:
+
+```python
+# requirements.txt
+
+torch
+transformers==3.0.*
+```
+
+<br>
+
+## Configure your API
+
+Create a `cortex.yaml` file and add the configuration below. A `SyncAPI` provides a runtime for inference and makes your `predictor.py` implementation available as a web service that can serve real-time predictions:
+
+```yaml
+# cortex.yaml
+
+- name: text-generator
+  kind: SyncAPI
+  predictor:
+    type: python
+    path: predictor.py
+```
+
+Here are the complete [API configuration docs](../../../docs/deployments/syncapi/api-configuration.md).
+
+<br>
+
+## Deploy your model locally
+
+`cortex deploy` takes your Predictor implementation along with the configuration from `cortex.yaml` and creates a web API:
+
+```bash
+$ cortex deploy
+
+creating text-generator (SyncAPI)
+```
+
+Monitor the status of your API using `cortex get`:
+
+```bash
+$ cortex get --watch
+
+env     sync api         status     last update   avg request   2XX
+local   text-generator   updating   8s            -             -
+```
+
+Show additional information for your API (e.g. its endpoint) using `cortex get <api_name>`:
+
+```bash
+$ cortex get text-generator
+status   last update   avg request   2XX
+live     1m            -             -
+
+endpoint: http://localhost:8888
+...
+```
+
+You can also stream logs from your API:
+
+```bash
+$ cortex logs text-generator
+
+...
+```
+
+Once your API is live, use `curl` to test your API:
```

Comment:
How long does this curl take? We should mention that it takes a few seconds if it takes a while.

Reply:
good call